flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerBaseTest.java

```diff
@@ -165,13 +164,13 @@ public void testAsyncErrorRethrownOnInvoke() throws Throwable {
             testHarness.processElement(new StreamRecord<>("msg-2"));
         } catch (Exception e) {
             // the next invoke should rethrow the async exception
-            Assert.assertTrue(e.getCause().getMessage().contains("artificial async exception"));
+            assertThat(e.getCause().getMessage()).contains("artificial async exception");
 
             // test succeeded
             return;
         }
 
-        Assert.fail();
+        fail("unknown failure");
```

Comment:
nit: Those are odd ways of doing it. I guess, using the assertThrows here would be more appropriate.